src/main/java/frc/robot/systems/EveryBotIntakeFSM.java

```diff
@@ -0,0 +1,340 @@
+	// FSM state definitions
+	public enum EveryBotIntakeFSMState {
+		INTAKING,
+		IDLE_FLIPCLOCKWISE,
+		IDLE_FLIPCOUNTERCLOCKWISE,
+		IDLE_STOP,
+		OUTTAKING
+	}
+	// Distance definitions
+	public enum ItemType {
+		CUBE,
+		CONE,
+		EMPTY
+	}
+	// hello
+	//HAVE TO CHANGE BASED ON TEST
+	private static final double KEEP_SPEED = 0;
+	private static final double INTAKE_SPEED = 0.1;
+	private static final double RELEASE_SPEED = -0.07; //DONT FORGET -
+	private static final double RELEASE_SPEED_LOW = -0.3;
+	private static final double CURRENT_THRESHOLD = 20;
+	private static final double BASE_THRESHOLD = 100;
+	private static final double TIME_RESET_CURRENT = 0.5;
+	private static final int MIN_RELEASE_DISTANCE = 800;
+	private static final int AVERAGE_SIZE = 10;
+	private static final double TURN_SPEED = -0.1;
+	private static final double OVERRUN_THRESHOLD = 0.007;
+	private static final double FLIP_THRESHOLD = 50;
+	//variable for armFSM, 0 means no object, 1 means cone, 2 means cube
+	private static ItemType itemType = ItemType.EMPTY;
+	private boolean isMotorAllowed = false;
+	private boolean toggleUpdate = true;
+	private boolean needsReset = true;
+	private int tick = 0;
+	private boolean hasTimerStarted = false;
+	private double[] currLogs = new double[AVERAGE_SIZE];
+
+
+	/* ======================== Private variables ======================== */
+	private EveryBotIntakeFSMState currentState;
+	// Hardware devices should be owned by one and only one system. They must
+	// be private to their owner system and may not be used elsewhere.
+	private CANSparkMax spinnerMotor;
+	private CANSparkMax flipMotor;
+	private Timer timer;
+	//private ElevatorArmFSM arm;
+	/* ======================== Constructor ======================== */
+	/**
+	 * Create FSMSystem and initialize to starting state. Also perform any
+	 * one-time initialization or configuration of hardware required. Note
+	 * the constructor is called only once when the robot boots.
+	 */
+	public EveryBotIntakeFSM() {
+		//arm = new ElevatorArmFSM();
+		timer = new Timer();
+		spinnerMotor = new CANSparkMax(HardwareMap.CAN_ID_SPINNER_MOTOR,
+				CANSparkMax.MotorType.kBrushless);
+		flipMotor = new CANSparkMax(HardwareMap.CAN_ID_FLIP_MOTOR,
+				CANSparkMax.MotorType.kBrushless);
+		// Reset state machine
+		reset();
+	}
+
+	/* ======================== Public methods ======================== */
+	/**
+	 * Return current FSM state.
+	 * @return Current FSM state
+	 */
+	public EveryBotIntakeFSMState getCurrentState() {
+		return currentState;
+	}
+	/**
+	 * Reset this system to its start state. This may be called from mode init
+	 * when the robot is enabled.
+	 *
+	 * Note this is distinct from the one-time initialization in the constructor
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = EveryBotIntakeFSMState.IDLE_STOP;
+		hasTimerStarted = false;
```

Comment:
Isn't this basically always true? Also what is this timer tracking? The name is a bit vague.